### ISSUE TITLE
survivalfly probem since build 1067

### ISSUE THREAD

**Issue by seema84** [NoCheatPlus/Issues#403]
#### Short description of the issue.
since build 1067 - latest survivalfly check dont work correctly
"every" player have a hight vl with player.survivalfly and have lags because of ncp

#### Complete output of the _ncp version_ command (please prefer text here).
build 1067 - 1072

#### How to reproduce the issue (is it random / always happens / side conditions)?
its random but often

#### Have you made any (noteworthy) changes to the default configuration of NCP?
not on survivalfly actions

#### Do you run any other special plugins that affect game mechanics next to NC+ (adding skills, machines, adding/changing blocks, other anti cheating)?
no

#### Provide additional information, such as error/debug/violation logs. (For reproducible issues with moving consider: https://github.com/NoCheatPlus/Docs/wiki/Debugging#on-the-fly-debug-output-for-individual-players) Use GitHub gists or http://pastebin.com/ or similar for huge logs, or attach a (zip-) file here. For image uploads, please prefer a site that allows viewing without JavaScript, you can upload file attachments for issues right here.

17-04-05 19:09:54 [INFO] Player failed SurvivalFly: tried to move: -4670.28, 67.04, 71.70 -> -4670.26, 67.16, 71.70, d=0.12 (climbdetached). VL 47.
17-04-05 19:13:41 [INFO] Player  failed SurvivalFly: tried to move: -4668.95, 69.52, 84.70 -> -4668.98, 69.64, 84.70, d=0.12 (climbdetached). VL 47.
17-04-05 19:18:31 [INFO] Player  failed SurvivalFly: tried to move: -4670.89, 75.02, 86.30 -> -4670.89, 75.42, 86.30, d=0.40 (vdistrel+ychincfly+vacc). VL 145.
17-04-05 20:51:05 [INFO] Player  failed SurvivalFly: tried to move: -4724.43, 100.02, 66.50 -> -4724.43, 100.42, 66.50, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.
17-04-05 20:57:29 [INFO] Player  failed SurvivalFly: tried to move: -2985.50, 57.02, 3335.09 -> -2985.50, 57.42, 3335.09, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.
17-04-05 21:30:48 [INFO] Player  failed SurvivalFly: tried to move: -4692.30, 64.02, 87.70 -> -4692.30, 64.42, 87.70, d=0.40 (dirty+vdistrel+ychincfly+vaccdirty). VL 131.
17-04-05 21:56:49 [INFO] Player  failed SurvivalFly: tried to move: -512.70, 65.02, 529.70 -> -512.70, 65.42, 529.69, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.

**Comment by asofold** [comment]
Could you add the (text) output of the 'ncp version' command? It's quite crucial to know the details (server versions, features, detected mc version, protocollib, other hooks...) - at least for one of your tests ?

Have you tested all builds from 1067 to 1072? Reference should be build 1072 or the latest build that has related changes rather, unless you want to distinguish towards a certain change (for future testing).

The shown violations don't look very severe  and they spread across 3 hours roughly. The logged details (thank you for using the defaults :) ) show something that may play a role in fixing some things, but it's hard to estimate if/how these are related to more severe problems.

So we need to start to distinguish:
* Players get frozen shortly after a false positive happens, e.g. after teleport, but they freeze slightly longer than before? (This could be "lags": Once they get pulled back, it shortly freezes them.)
* Can anything specific be told otherwise?

(Explanation: The recent changes are necessary to prevent other plugins to assume NCP setting back players is normal feature-based teleporting, due to a recent change in the spigot server. Unfortunately this forces us to freeze the player until the player has been teleported to the set back location, which can lead to freezing the player for at least the round-trip-time, if the players are moving. 250 ms ping -> 0.5 seconds freeze, roughly - it's still possible that there are edge cases or bugs to consider, before going for complicated workarounds or pull requests to change the server to allow us immediate set backs again.)

**Comment by seema84** [comment]
[00:36:18 INFO]: ---- Version information ----
[00:36:18 INFO]: #### Server ####
[00:36:18 INFO]: git-Spigot-625bc00-b5f7b6d ~MC: 1.11.2~
[00:36:18 INFO]:   detected: 1.11.2
[00:36:18 INFO]: #### NoCheatPlus ####
[00:36:18 INFO]: Plugin: 3.15.0-SNAPSHOT-sMD5NET-b1072
[00:36:18 INFO]: MCAccess: 1.11-1.11.2 / Spigot-CB-1.11_R1
[00:36:18 INFO]: Features:
[00:36:18 INFO]:   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9 | BlocksMC1_10 | BlocksMC1_11
[00:36:18 INFO]:   checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter
[00:36:18 INFO]:   defaults: pvpKnockBackVelocity
[00:36:18 INFO]: Hooks: AllViolations~NCP~ 1.0

I have tested all Builds from 1067 to 1072.

Yes, players get frozen shortly after a false positive happens (vertical and horizontally).
One of my tested player has a 200.000 dsl connection with ~20 ms ping.

**Comment by asofold** [comment]
Please check out build 1073 for comparison.

**Comment by seema84** [comment]
1073 dont fix this problem.

**Comment by seema84** [comment]
directly after restart with this new build ->

17-04-06 20:09:39 [INFO] Player failed SurvivalFly: tried to move: -4739.42, 105.02, 49.38 -> -4739.42, 105.42, 49.38, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.
17-04-06 20:23:10 [INFO] Player failed SurvivalFly: tried to move: -4773.71, 83.02, 59.62 -> -4773.71, 83.42, 59.62, d=0.40 (vdistrel+ychincfly+vacc). VL 145.
17-04-06 20:48:47 [INFO] Player2 failed SurvivalFly: tried to move: 2349.63, 68.02, -1373.68 -> 2349.63, 68.02, -1373.68, d=0.00 (lowjump+vacc). VL 37.
17-04-06 20:49:03 [INFO] Player2 failed SurvivalFly: tried to move: 2337.44, 68.02, -1365.70 -> 2337.44, 68.02, -1365.70, d=0.00 (lowjump+vacc). VL 27.
17-04-06 20:49:15 [INFO] Player2 failed SurvivalFly: tried to move: 2341.52, 68.02, -1361.43 -> 2341.52, 68.02, -1361.43, d=0.00 (dirty+vaccdirty). VL 40.
17-04-06 20:52:33 [INFO] Player2 failed SurvivalFly: tried to move: 2349.53, 66.02, -1360.66 -> 2349.53, 66.42, -1360.66, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 171.
17-04-06 21:00:09 [INFO] Player2 failed SurvivalFly: tried to move: 2341.81, 66.02, -1360.37 -> 2341.81, 66.42, -1360.37, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.
17-04-06 21:05:02 [INFO] Player2 failed SurvivalFly: tried to move: 2336.87, 66.02, -1374.30 -> 2336.87, 66.42, -1374.30, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.
17-04-06 21:40:50 [INFO] Player failed SurvivalFly: tried to move: -4651.73, 70.02, 2.35 -> -4651.73, 70.42, 2.35, d=0.40 (vdistrel+lowjump+ychincfly+vacc). VL 145.
17-04-06 21:51:27 [INFO] Player failed SurvivalFly: tried to move: -4733.55, 73.05, 45.31 -> -4733.17, 73.47, 45.60, d=0.64 (bunnyenv+permchecks+bunnyenv+hbufuse+vdistrel+lowjump+ychincfly+vacc). VL 145.

**Comment by asofold** [comment]
* Are issues exactly like before, concerning frequency and severity?
* If it's easy to reproduce, have you considered creating a debug log for a certain player? https://github.com/NoCheatPlus/Docs/wiki/Debugging#on-the-fly-debug-output-for-individual-players
* Are other anti cheating plugins in use, or other plugins correcting positions of players (idk perhaps on teleporting or so, we've had custom plugins do such for instance)?

**Comment by seema84** [comment]
1. yes
2. i have don't debug this problem
3. no

**Comment by asofold** [comment]
(2.) We need to track down further...
* Quality - what kind of things happen with 1073.
* Quantity - how often do they happen for one player.
* Confine side conditions - sprint-jumping? Special blocks nearby? A flat world does?
(3.) Just to ensure: Add-ons to NCP that alter set back handling could also conflict.

I'll also test with a remote server with simulated load/lag more. So far on a local test server i couldn't get much noticable except one passable-set-back-loop with jumping onto the edge of a block, which probably is fixed by now. I also test with speed-up boots (+0.5 speed is hilarious for instance) also on randomly layered snow and special blocks to sprint/jump over... but for a local test server i don't manage to create violations during "only run throughthe world in 2 min".

**Comment by seema84** [comment]
"Quality - what kind of things happen with 1073."
The same as build 1067 - 1072.

"Quantity - how often do they happen for one player."
In my test 3 times in 2 Minutes.

"Confine side conditions - sprint-jumping? Special blocks nearby? A flat world does?"
Not a flat world. I test it on a normal survival world. I get this issue on sprint jump, normal flat sprint run and build vertical (non sprint). 

"(3.) Just to ensure: Add-ons to NCP that alter set back handling could also conflict."
I don't use any NCP add-ons. 

"4. ViaVersion or BungeeCord in use?"
No

**Comment by asofold** [comment]
Build 1074 contains further changes, such as
* skip unnecessary teleporting (current location or packet level indicate it's received).
* accept positions as set back, disregarding looking direction. Not sure this is relevant, though.

**Comment by seema84** [comment]
Build 1074 don't fix this issue. After i restart the server with this build and run 20 seconds i get one survivalfly vl.

**Comment by asofold** [comment]
False positives in survivalfly can still happen, they are not altered by recent changes per se - only what happens with/after a false positive, that has changed (e.g. longer freezing, getting stuck perhaps, such). So if you run off spawn and get a violation,  i can't deduce anything in relation to the recent changes from this :).

Now the question is, if you get "rubberbanded" for a longer time than with build 1066. If you could produce a debug log, especially if it's so easy to reproduce for you, that could really help (best reproduce several times, in case there is differing side conditions). Especially on 1.11.2 with default spigot, i don't get _any_ violations with just running somewhere (remote server testing still pending).

**Comment by seema84** [comment]
here is a debug log where i get "one" issue https://pastebin.com/N9vPBbt8

**Comment by asofold** [comment]
According to the log, the set back handling works as swift as possible, so i assume this is just a false positive, with no difference to build 1066. Provided there is no extra bugs leading to more false positives.

You don't have ProtocolLib installed , though. With ProtocolLib we would see slightly more details, such as incoming flying packets and outgoing position packets. 

At least we can see on Bukkit side, that there are no further moving events cancelled, so it does look like an ordinary false positive with one time set back.

Did you experience that set back different to with build 1066? Did you get frozen longer?

**Comment by seema84** [comment]
"Did you experience that set back different to with build 1066? Did you get frozen longer?"
no same frozen time (very short)

I can try to debug with protocollib if it helps.

**Comment by asofold** [comment]
Yes, that would be good!

**Comment by seema84** [comment]
https://www.file-upload.net/download-12424773/nocheatplus.log.html
again with one issue

**Comment by asofold** [comment]
The log still looks like a very swift set back. I can't see a difference to older versions of NCP, concerning the way the set back works (e.g. extra freezing).

The issue why you are set back is another thing, this may be fixable independent of recent changes.

----

By the way, did you know you can upload any text as gist here (https://gist.github.com/), or images/text/zip files just as files (with a new or editing a comment, below the blue dashed line on the bottom of the edit area)? That would be the most simple way to add files or link readable text, unless you want to ensure it's stored on certain servers other than github.

**Comment by seema84** [comment]
no i don't know (gist), thanks

but what can i now do to help you to fix my issue? i can create a video but i don't think it helps

**Comment by seema84** [comment]
i have test it on a fresh server (local on my pc) only with ncp but this issue stay
with build 1066 it works fine

**Comment by asofold** [comment]
You provided the debug logs which will help track it down, regardless what it is caused by.

If you have too much time :), you could try to make it 100% reproducible with an impossible to fail setup (map + how to start), and provide debug logs for both the latest NCP (now: 1074) and for NCP 1066, while the latter of it doesn't have the violation+set back in it (use the action definitions of build 1074).

E.g. if this is reproducible this way (only if), using ncp 1074, preferably as the only plugin:
* create a flat world (no height differences).
* use 'ncp remove *' to ensure all data is reset. This also resets the horizontal buffer, which is used for leniency.
* start debug logging, uinless your test server has it set in the configuration (dont mix up order with remove).
* (Walk around slowly for a bit to ensure data is really initialized and the leniency buffer is filled. Could leave this for the second run, if this all works.)
* Start sprint-jumping into one direction (no obstacles) - until the set back occurs.
* (Don't do it all night, for the case that other side conditions are involved :p.)

If this produces a false positive, you could do the exact same setup with build 1066, ensure to run the same distance. If you know the distance, you could also use the ncp stopwatch distance ... command using the distance + idk 50 or 100 for comparison. A slightly shorter run  might do, because earlier on the recorded data will show a difference. Optimal would be the same amount of distance/time for both logs.

**I don't demand you to do this / anything else**, given what you already posted, i'll likely be able to do something. Honestly, given issues with (supposedly) simple bunny-hopping, i will try to reproduce the same way. Ideally this is something that a team member could do (...), but for that purpose we'd need people who can do this kind of thing. Actually i'll ask @MyPictures to do this, hehe, he's really good at reproducing issues.

**Comment by MyPictures** [comment]
So I did a test with NC+ 1074 (with ProtocolLib):
[21:23:10 INFO]: ---- Version information ----
[21:23:10 INFO]: #### Server ####
[21:23:10 INFO]: git-Spigot-d276ab1-d219213 ~MC: 1.11.2~
[21:23:10 INFO]:   detected: 1.11.2
[21:23:10 INFO]: #### NoCheatPlus ####
[21:23:10 INFO]: Plugin: 3.15.0-SNAPSHOT-sMD5NET-b1074
[21:23:10 INFO]: MCAccess: 1.11-1.11.2 / Spigot-CB-1.11_R1
[21:23:10 INFO]: Features:
[21:23:10 INFO]:   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9 | BlocksMC1_10 | BlocksMC1_11
[21:23:10 INFO]:   checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter | AttackFrequency | FlyingFrequency | KeepAliveFrequency
[21:23:10 INFO]:   defaults: pvpKnockBackVelocity
[21:23:10 INFO]:   packet-listeners: DebugAdapter | UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
[21:23:10 INFO]: Hooks: AllViolations~NCP~ 1.0 | TestNCPHook 1.0
[21:23:10 INFO]: #### Related Plugins ####
[21:23:10 INFO]: ProtocolLib v4.2.1

Seems like bunny hopping (as it has always been) + lost on ground?

[17-04-10-1.txt](https://github.com/NoCheatPlus/Issues/files/911210/17-04-10-1.txt)

**Comment by asofold** [comment]
@MyPictures Thank you :). It doesn't look like lost ground (do you see that anywhere!?) - it rather looks like there is lowjump in between and "bunny loss", i.e. NCP doesn't accept it to be bunny hopping, so it starts consuming the horizontal buffer of the player.

This was a flat world?

**Comment by MyPictures** [comment]
@asofold no it happend on mountains but cannot fully remember when exactly.

**Comment by MyPictures** [comment]
Here is another one asofold on a flat world running the same setup as above.

[17-04-11-1.txt](https://github.com/NoCheatPlus/Issues/files/913949/17-04-11-1.txt)

**Comment by asofold** [comment]
The actions entries for survivalfly and creativefly have been faulty since build 1067. Build 1080 contains a fix for that. Alternatively you could use build 1075 and manually correct the actions ('cancelvl' should be 'cancel vl').

Otherwise i'll tackle the sprintjump issues very soon.

**Comment by oloflarsson** [comment]
I'm also seeing a lot of false survivalfly positives. Running 3.15.0-SNAPSHOT-sMD5NET-b1081. I'll downgrade and see if it happens.

```
---- Version information ----
#### Server ####
git-Paper-1081 ~MC: 1.11.2~
  detected: 1.11.2
#### NoCheatPlus ####
Plugin: 3.15.0-SNAPSHOT-sMD5NET-b1081
MCAccess: 1.11-1.11.2 / Spigot-CB-1.11_R1
Features:
  blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9 | BlocksMC1_10 | BlocksMC1_11
  checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter | AttackFrequency | FlyingFrequency | KeepAliveFrequency
  defaults: pvpKnockBackVelocity
  packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
Hooks: AllViolations~NCP~ 1.0 | Citizens2~cncp~ 2.0 | MassiveLock 2.11-SNAPSHOT | MassiveMagic 2.11-SNAPSHOT | mcMMO~cncp~ 2.3
#### Related Plugins ####
CompatNoCheatPlus v6.6.3-SNAPSHOT-sMD5NET-b87 | ProtocolLib v4.2.1
```

**Comment by seema84** [comment]
build #1084 don't solved this issue

**Comment by asofold** [comment]
@oloflarsson You have a couple of special hooks (Massive...), do they involve skills like special jumping and the like?

**Comment by i9hdkill** [comment]
`#### Server ####
git-Spigot-d276ab1-1f6c04c ~MC: 1.11.2~
  detected: 1.11.2
#### NoCheatPlus ####
Plugin: 3.15.2-SNAPSHOT-sMD5NET-b1089
MCAccess: 1.11-1.11.2 / Spigot-CB-1.11_R1
Features:
  blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9 | BlocksMC1_10 | BlocksMC1_11
  checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter | AttackFrequency | FlyingFrequency | KeepAliveFrequency
  defaults: pvpKnockBackVelocity
  packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
Hooks: AllViolations~NCP~ 1.0
#### Related Plugins ####
ProtocolLib v4.2.1`

1.9-1.11.2 players are getting kicked when jumping on lillypads with NCP Build 1089

https://pastebin.com/HzUGqUmk

**Comment by asofold** [comment]
@i9hdkill Despite the overly broad issue title, this is another issue (!). Do you have ViaVersion installed ? Othewise which protocol support plugin/technique are you using?

This may be your issue: https://github.com/NoCheatPlus/Issues/issues/274
The current solution is to add an entry to compatibility.blocks.overrideflags:
water_lily: default+ground+ground_height+height8_1

I'll a a default for if ViaVersion is installed so perhpas changing the configuration won't be necessary from there on.

**Comment by seema84** [comment]
no improvement since the issue creation

**Comment by asofold** [comment]
Closing as superseeded - too many changes since then. If it stays an issue we'll need to reevaluate:
* Latest build(s) of NCP.
* ncp version
* Plugin list (related ~ skills, anti cheating, altering moving/teleport/block-shapes, protocol support, ...).
* If possible, confine to side conditions. A debug log of issues would be great.
